src/devices/efergy_e2_classic.c

```diff
@@ -60,24 +68,59 @@ static int efergy_e2_classic_callback(bitbuffer_t *bitbuffer) {
 		return 0;
 	}
 
-	const unsigned VOLTAGES[] = {110, 115, 120, 220, 230, 240, 0};
-
-	double current_adc = 256 * bytes[4] + bytes[5];
-	for (unsigned i = 0; VOLTAGES[i] != 0; ++i) {
-		double power  = (VOLTAGES[i] * current_adc * (1 << bytes[6])) / 32768;
-		fprintf(stderr, "Power consumption at %u volts: %.2f watts\n", VOLTAGES[i], power);
+	if (bytes[6] > 0x80)
```

Comment:
Mmmh, will not that convert the value of bytes[6] (uint8_t) to an int8_t (keeping the same positive value then)? Shouldn't I reinterpret bytes[6] as a signed number? (ex: using reinterpret_cast<int8_t>(bytes[6])). For memo, fact can be negative to shift the comma to the left.
I haven't been developing in C for a while to be honest... ;)